src/scenic/core/pruning.py

```diff
@@ -380,79 +396,125 @@ def pruneVisibility(scenario, verbosity):
     for obj in scenario.objects:
         # Extract the base region if it exists
         position = currentPropValue(obj, "position")
-        base, offset = matchInRegion(position)
+        base, offset, pir_dist = matchInRegion(position)
 
-        if base is None or needsSampling(base):
+        # Compute the maximum distance the object can be from the sampled point
+        if offset is not None:
+            _, maxDistance = supportInterval(offset.norm())
+        else:
+            maxDistance = 0
+
+        if (
+            base is None
+            or needsSampling(base)
+            or needsSampling(obj.radius)
+            or maxDistance is None
+        ):
             continue
 
-        newBase = base
+        currBase = base
+        currDist = pir_dist
+        currPos = position
 
-        # Define a helper function to buffer an oberver's visibleRegion, resulting
+        # Define a helper function to attempt buffer an oberver's visibleRegion, resulting
         # in a region that contains all points that could feasibly be the position
-        # of obj, if it is visible from the observer.
+        # of obj, if it is visible from the observer. If possible buffer exactly, otherwise
+        # try to buffer approximately, and if that is also not feasible just return the viewRegion.
         def bufferHelper(viewRegion):
-            return viewRegion._bufferOverapproximate(obj.radius / 2, PRUNING_PITCH)
+            buffer_quantity = obj.radius / 2 + maxDistance
```

Comment:
Seems like this should be just `obj.radius`, not half of that (the position could be as far off as `obj.radius` and still have part of the object be in the visible region).

Reply:
Good catch!